<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ActuatorWidget</class>
 <widget class="QWidget" name="ActuatorWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>545</width>
    <height>48</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="toolTip">
   <string>High Trim</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_3">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QStackedWidget" name="stackedWidgetViews">
     <property name="sizePolicy">
      <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="pageSettings">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QPushButton" name="pushButtonLimp">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Limp actuators will not have power and move freely</string>
         </property>
         <property name="text">
          <string>Limp</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButtonCenter">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Set actuator target position to center value.</string>
         </property>
         <property name="text">
          <string>Center</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBoxReversed">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Reverse actuator motion direction.</string>
         </property>
         <property name="text">
          <string>Reversed</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="doubleSpinBoxLowTrim">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Low trim</string>
         </property>
         <property name="accelerated">
          <bool>true</bool>
         </property>
         <property name="correctionMode">
          <enum>QAbstractSpinBox::CorrectionMode::CorrectToNearestValue</enum>
         </property>
         <property name="showGroupSeparator" stdset="0">
          <bool>false</bool>
         </property>
         <property name="decimals">
          <number>3</number>
         </property>
         <property name="minimum">
          <double>-1.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="doubleSpinBoxHighTrim">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string>High trim</string>
         </property>
         <property name="accelerated">
          <bool>true</bool>
         </property>
         <property name="correctionMode">
          <enum>QAbstractSpinBox::CorrectionMode::CorrectToNearestValue</enum>
         </property>
         <property name="showGroupSeparator" stdset="0">
          <bool>false</bool>
         </property>
         <property name="decimals">
          <number>3</number>
         </property>
         <property name="minimum">
          <double>-1.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButtonSetDefault">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Choose current target position as center value.</string>
         </property>
         <property name="text">
          <string>Set Default</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Policy::MinimumExpanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pageTarget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QCheckBox" name="checkBoxEnabled">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Only Enabled actuators can be controlled by Agent.</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelIndex">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>10</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>28</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Actuator index</string>
         </property>
         <property name="text">
          <string>000</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
         </property>
         <property name="margin">
          <number>3</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="EditableLabel" name="editableLabelName">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Actuator Name</string>
         </property>
         <property name="text">
          <string>#</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="DecimalEntry" name="decimalEntryServoPosition" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Actuator position</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="pushButtonSettings">
     <property name="minimumSize">
      <size>
       <width>30</width>
       <height>30</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>30</width>
       <height>30</height>
      </size>
     </property>
     <property name="text">
      <string>X</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>EditableLabel</class>
   <extends>QLabel</extends>
   <header location="global">components/EditableLabel.hpp</header>
  </customwidget>
  <customwidget>
   <class>DecimalEntry</class>
   <extends>QWidget</extends>
   <header location="global">components/DecimalEntry.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButtonLimp</sender>
   <signal>toggled(bool)</signal>
   <receiver>ActuatorWidget</receiver>
   <slot>toggleLimp(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>508</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxEnabled</sender>
   <signal>toggled(bool)</signal>
   <receiver>ActuatorWidget</receiver>
   <slot>toggleEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>125</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxReversed</sender>
   <signal>toggled(bool)</signal>
   <receiver>ActuatorWidget</receiver>
   <slot>toggleReversed(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>350</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxHighTrim</sender>
   <signal>valueChanged(double)</signal>
   <receiver>ActuatorWidget</receiver>
   <slot>highTrimUpdated(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>672</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxLowTrim</sender>
   <signal>valueChanged(double)</signal>
   <receiver>ActuatorWidget</receiver>
   <slot>lowTrimUpdated(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>271</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonSettings</sender>
   <signal>toggled(bool)</signal>
   <receiver>ActuatorWidget</receiver>
   <slot>toggleSettings(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1397</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>706</x>
     <y>23</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>toggleEnabled(bool)</slot>
  <slot>toggleReversed(bool)</slot>
  <slot>toggleLimp(bool)</slot>
  <slot>lowTrimUpdated(double)</slot>
  <slot>highTrimUpdated(double)</slot>
  <slot>toggleSettings(bool)</slot>
 </slots>
</ui>
